src/compiler/checker.ts

```diff
@@ -34744,6 +34744,9 @@ namespace ts {
                         Diagnostics.const_enum_member_initializer_was_evaluated_to_a_non_finite_value);
                 }
             }
+            else if (initializer.kind === SyntaxKind.Identifier && (initializer as Identifier).escapedText === "undefined") {
+                return undefined;

```

Comment:
Shouldn't there be an error here? I thought `undefined` was supposed to be an error, but with an undefined-specific message.